core/primitives/src/views.rs

```diff
@@ -1416,6 +1416,8 @@ pub struct CurrentEpochValidatorInfo {
     pub shards: Vec<ShardId>,
     pub num_produced_blocks: NumBlocks,
     pub num_expected_blocks: NumBlocks,
+    pub num_produced_chunks: NumBlocks,
+    pub num_expected_chunks: NumBlocks,
```

Comment:
This is a `Borsh` structure, so adding fields is generally breaking (it changes offsets of everything). However, it seems that this is BorsSerialize accidently. Let's apply the following diff to make sure borsh isn't a concern: 

```diff
diff --git a/core/primitives/src/views.rs b/core/primitives/src/views.rs
index b45b9c7bc..73d531f32 100644
--- a/core/primitives/src/views.rs
+++ b/core/primitives/src/views.rs
@@ -1378,7 +1378,7 @@ impl TryFrom<ReceiptView> for Receipt {
 
 /// Information about this epoch validators and next epoch validators
 #[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
-#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct EpochValidatorInfo {
     /// Validators for the current epoch
     pub current_validators: Vec<CurrentEpochValidatorInfo>,
@@ -1406,7 +1406,7 @@ pub struct ValidatorKickoutView {
 }
 
 #[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
-#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct CurrentEpochValidatorInfo {
     pub account_id: AccountId,
     pub public_key: PublicKey,
```

Though, even for JSON, what if an RPC client which uses new version contracts a near node which is still at the old version? I think the client would  fail to deserialize this. So maybe let's also `#[serde(default)]`?

Reply:
Oof good catch. Updated.

Reply:
Also on this note, I wonder if https://github.com/near/nearcore/pull/4221 broke anything? I would guess the same issues would happen there, but maybe nobody noticed :/

Reply:
I think not, because, as far as I know, the borsh impl actually unused.